Residual drift from #1114 in ai-sdk@0.5.0: `ReflectBasedOn.mental_models` inline shape still disagrees with `hindsight-client@0.5.3` + OpenAPI (`{id, text, context?}` vs `{id, name, content?}`)

## Summary

[#1118](https://github.com/vectorize-io/hindsight/pull/1118) (merged 2026-04-17, shipped as `@vectorize-io/hindsight-ai-sdk@0.5.0` + `@vectorize-io/hindsight-client@0.5.3`) closed [#1114](https://github.com/vectorize-io/hindsight/issues/1114) and aligned most of the `HindsightClient` interface with the real client class. **However, one residual drift was left behind**: the inline type for `ReflectBasedOn.mental_models` inside `hindsight-ai-sdk` still uses the *wrong field names*, so `createHindsightTools({ client })` still rejects a plain `HindsightClient` instance without an `as unknown as HindsightClient` cast, and any consumer that reads `based_on.mental_models[].name` / `.content` gets `undefined` at runtime.

Looks like the mental-model shape was hand-typed using the fields from the top-level `MentalModelResponse` (which IS `{ id, name, content }`), but the nested `ReflectMentalModel` the server actually emits is `{ id, text, context? }`.

## Versions

- `@vectorize-io/hindsight-ai-sdk@0.5.0` (latest)
- `@vectorize-io/hindsight-client@0.5.3` (latest)
- `ai@^6.0.168`

## The drift

Three sources of truth, two of which agree:

| Source | `ReflectMentalModel` shape |
|---|---|
| Server OpenAPI (`openapi.json`, ground truth) | `{ id: string, text: string, context?: string \| null }` ✅ |
| `@vectorize-io/hindsight-client@0.5.3` (`dist/index.d.ts`) | `{ id: string, text: string, context?: string \| null }` ✅ matches server |
| `@vectorize-io/hindsight-ai-sdk@0.5.0` — `ReflectBasedOn.mental_models` inline | `{ id: string, name: string, content?: string \| null }` ❌ wrong field names |

Concretely, from `hindsight-integrations/ai-sdk/src/tools/index.ts` on `main`:

\`\`\`ts
export interface ReflectBasedOn {
  memories?: ReflectFact[];
  mental_models?: Array<{ id: string; name: string; content?: string | null }>;  // ❌
  directives?: Array<{ id: string; content: string }>;                            // also missing \`name\`
}
\`\`\`

And from `hindsight-client@0.5.3` (which mirrors the OpenAPI spec):

\`\`\`ts
type ReflectMentalModel = {
  id: string;
  text: string;
  context?: string | null;
};
\`\`\`

## Runtime impact

The ai-sdk's `reflect` tool passes `result.based_on` through unchanged:

\`\`\`js
// hindsight-ai-sdk/dist/index.js (reflect tool)
return {
  text: result.text ?? "No insights available yet.",
  basedOn: result.based_on
};
\`\`\`

So consumers who follow the SDK's types see:

\`\`\`tsx
{basedOn.mental_models?.map(m => <div>{m.name}</div>)}
// TypeScript happy, runtime renders undefined — server sent m.text
\`\`\`

## Also worth fixing while we're in there

`ReflectDirective` per OpenAPI has `{ id, name, content, ... }`, but the SDK inline type omits `name`. Less harmful (no swapped fields), but still a drift.

## TypeScript impact

Because the SDK's declared `HindsightClient.reflect()` return type says `mental_models[]: { id, name, content? }` and the real `HindsightClient` class returns `mental_models[]: { id, text, context? }`, structural assignability still fails:

\`\`\`
Types of property 'mental_models' are incompatible.
  Type 'ReflectMentalModel[] | undefined' is not assignable to
       type '{ id: string; name: string; content?: string | null }[] | undefined'.
    Property 'name' is missing in type 'ReflectMentalModel'.
\`\`\`

So users still have to write:

\`\`\`ts
createHindsightTools({
  client: hindsightClient as unknown as HindsightClient, // still required after #1118
});
\`\`\`

— which defeats the stated goal of #1118 ("eliminate the need for the cast / make the README example actually compile").

## Reproduction

\`\`\`ts
import { HindsightClient } from "@vectorize-io/hindsight-client"; // 0.5.3
import { createHindsightTools } from "@vectorize-io/hindsight-ai-sdk"; // 0.5.0

const client = new HindsightClient({ baseUrl: "http://localhost:8888" });
// TS error TS2345: mental_models[].name missing in ReflectMentalModel
const tools = createHindsightTools({ client, bankId: "b" });
\`\`\`

## Expected fix

In `hindsight-integrations/ai-sdk/src/tools/index.ts`, replace the inline mental-model shape (and, ideally, align `ReflectFact` / `ReflectDirective` too) so the SDK's `ReflectBasedOn` uses the same shapes the server/OpenAPI actually emit:

\`\`\`ts
export interface ReflectBasedOn {
  memories?: ReflectFact[];
  mental_models?: Array<{ id: string; text: string; context?: string | null }>;
  directives?: Array<{ id: string; name?: string; content: string }>;
}
\`\`\`

Or, cleaner, import `ReflectMentalModel` / `ReflectDirective` from `@vectorize-io/hindsight-client` (or the shared OpenAPI-generated types) so the two packages can't drift again.

After that, the README example compiles cast-free and consumers' runtime access to `mental_models[].text` matches what the LLM actually received.

## Context

Filed as a follow-up to #1114 / #1118 (closed) since those specifically claim to eliminate the cast, and users on `^0.5.0` / `^0.5.3` (including us) are still carrying it.

--- a/hindsight-integrations/ai-sdk/src/tools/index.ts
+++ b/hindsight-integrations/ai-sdk/src/tools/index.ts
@@ -76,8 +76,8 @@ export interface ReflectFact {
  */
 export interface ReflectBasedOn {
   memories?: ReflectFact[];
-  mental_models?: Array<{ id: string; name: string; content?: string | null }>;
-  directives?: Array<{ id: string; content: string }>;
+  mental_models?: Array<{ id: string; text: string; context?: string | null }>;
+  directives?: Array<{ id: string; name: string; content: string }>;
 }
 
 /**

